pynsee/utils/requests_session.py

```diff
@@ -0,0 +1,360 @@
+        # Unused (managed through a specific if/else), kept for memory purpose
+        # 429: "Too Many Requests : allocated quota overloaded",
+        500: "Internal Server Error ",
+        503: "Service Unavailable",
+    }
+
+    def __init__(
+        self,
+        http_proxy: str = None,
+        https_proxy: str = None,
+        sirene_key: str = None,
+    ):
+        """
+        Generates a PynseeAPISession used to perform http(s) requests in pynsee
+
+        If the instance is created without arguments, pynsee will try to get
+        variables from environment; if that fails, it will try to retrieve the
+        latest config (stored on disk).
+
+        In the other case (with at least one not-None argument), credentials
+        stored on disk will be ignored and proxies/sirene key arguments will
+        use said arguments.
+        **This should only be done through pynsee.utils.init_conn**.
+
+        See pynsee.utils.init_conn to create a new configuration.
+
+        Parameters
+        ----------
+        http_proxy : str, optional
+            HTTP proxy. Use this if working behind a corporate proxy and pynsee
+            has not already been configured on the machine. Otherwise, leave
+            this to None. The default is None.
+        https_proxy : str, optional
+            HTTPS proxy. Use this if working behind a corporate proxy and
+            pynsee has not already been configured on the machine. Otherwise,
+            leave this to None. The default is None.
+        sirene_key : str, optional
+            Token generated by the API portal. Use this if you wish to access
+            the SIRENE API and pynsee has not already been configured. The
+            default is None.
+
+        """
+
+        super().__init__()
+        self._mount_adapters()
+
+        if any(x is not None for x in (http_proxy, https_proxy, sirene_key)):
+            # set credentials for a new pynsee connection
+            credentials = {
+                "http_proxy": http_proxy,
+                "https_proxy": https_proxy,
+                "sirene_key": sirene_key,
+            }
+        else:
+            credentials = _get_credentials()
+
+        proxies = {
+            "http": credentials.get("http_proxy"),
+            "https": credentials.get("https_proxy"),
+        }
+        self.proxies.update(proxies)
+
+        username = os.environ.get("USERNAME", "username")
+        version = pynsee.__version__
+        useragent = {"User-Agent": f"python_pynsee_{username}/{version}"}
+
+        # Note : geoplatform seems to impose the "/version" to the user-agent
+        self.headers.update(useragent)
+
+        self.sirene_key = credentials.get("sirene_key", None)
+        self.headers["X-INSEE-Api-Key-Integration"] = self.sirene_key
+
+    def _mount_adapters(self):
+
+        # default retries adapter
+        retry_adapt = Retry(
+            total=7, backoff_factor=1, status_forcelist=[429, 502, 503, 504]
```

Comment:
could you add the meaning of the missing codes in the list at the beginning of the class?

Reply:
No, the codes at the beginning are from the API portal's documentations. The other here are standard codes. But I can add those translations here of course.